services/workflows-service/src/business-report/merchant-monitoring-client.ts

```diff
@@ -0,0 +1,221 @@
+import { Injectable } from '@nestjs/common';
+import { z } from 'zod';
+import axios, { AxiosInstance } from 'axios';
+import { env } from '@/env';
+import { CountryCode } from '@/common/countries';
+import {
+  MERCHANT_REPORT_STATUSES,
+  MERCHANT_REPORT_TYPES,
+  MERCHANT_REPORT_TYPES_MAP,
+  MERCHANT_REPORT_VERSIONS,
+  MerchantReportType,
+  MerchantReportVersion,
+} from '@/business-report/constants';
+import { TReportRequest } from '@/common/utils/unified-api-client/unified-api-client';
+
+const CreateReportResponseSchema = z.object({});
+const CreateReportBatchResponseSchema = z.array(
+  z.object({
+    reportId: z.string(),
+  }),
+);
+const ReportSchema = z.object({
+  id: z.string(),
+  websiteId: z.string(),
+  merchantId: z.string(),
+  reportType: z.enum([MERCHANT_REPORT_TYPES[0]!, ...MERCHANT_REPORT_TYPES.slice(1)]),
+  workflowVersion: z.enum([MERCHANT_REPORT_VERSIONS[0]!, ...MERCHANT_REPORT_VERSIONS.slice(1)]),
+  parentCompanyName: z.string().nullable(),
+  status: z.enum([MERCHANT_REPORT_STATUSES[0]!, ...MERCHANT_REPORT_STATUSES.slice(1)]),
+  metadata: z
+    .object({
+      workflowRuntimeDataId: z.string().optional(),
+    })
+    .nullable(),
+  riskScore: z.number().nullable(),
+  isAlert: z.boolean().nullable(),
+  website: z.object({
+    id: z.string(),
+    url: z.string().url(),
+    createdAt: z
+      .string()
+      .datetime()
+      .transform(value => new Date(value)),
+    updatedAt: z
+      .string()
+      .datetime()
+      .transform(value => new Date(value)),
+  }),
+  createdAt: z
+    .string()
+    .datetime()
+    .transform(value => new Date(value)),
+  updatedAt: z
+    .string()
+    .datetime()
+    .transform(value => new Date(value)),
+  data: z.record(z.string(), z.unknown()).nullish(),
+});
+const FindManyReportsResponseSchema = z.object({
+  totalItems: z.number(),
+  totalPages: z.number(),
+  data: z.array(ReportSchema),
+});
+
+@Injectable()
+export class MerchantMonitoringClient {
+  private axios: AxiosInstance;
+
+  constructor() {
+    this.axios = axios.create({
```

Comment:
might be a good place to enforce having a customer attached to request - or even better use the unified API service and enforce it here, in one consolidated place